### PR TITLE
Update google analytics search events

### PR DESCRIPTION
**Pull request by yurm04**
### WHY are these changes introduced?
In order to capture custom data with GAUA we need to define a custom map for metrics. See [GAUA docs for how we're doing this](https://developers.google.com/analytics/devguides/collection/gtagjs/custom-dims-mets)

### WHAT is this pull request doing?
* removing `Global Search` event because we don't need it anymore
* adding custom map for custom metrics
* updating custom metric names to match conventions

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris.shopify.com/pages/_app.tsx [hunk 1]
```diff
@@ -13,7 +13,14 @@ import '../src/styles/globals.scss';
 const PUBLIC_GA_ID = 'UA-49178120-32';
 
 const gaPageView = (url: string) => {
-  window.gtag('config', PUBLIC_GA_ID, {page_path: url});
+  window.gtag('config', PUBLIC_GA_ID, {
+    page_path: url,
+    custom_map: {
```
Comment (yurm04) [inline polaris.shopify.com/pages/_app.tsx:18]: following the docs here https://developers.google.com/analytics/devguides/collection/gtagjs/custom-dims-mets

File: polaris.shopify.com/src/components/GlobalSearch/GlobalSearch.tsx [hunk 2]
```diff
@@ -52,23 +52,14 @@ function captureSearchEvent(
...
-    searchTerm,
-    resultRank,
-    selectedResult,
+    search_term: searchTerm,
+    result_rank: resultRank,
+    selected_result: selectedResult,
     category: 'engagement',
   };
 
-  const googleParams = {
```
Comment (yurm04) [inline polaris.shopify.com/src/components/GlobalSearch/GlobalSearch.tsx:61]: we only need one of these events and the custom event is set up with the metrics we want

File: polaris.shopify.com/src/components/GlobalSearch/GlobalSearch.tsx [hunk 3]
```diff
@@ -52,23 +52,14 @@ function captureSearchEvent(
   if (!searchTerm) return;
 
   const customParams = {
-    searchTerm,
-    resultRank,
-    selectedResult,
+    search_term: searchTerm,
```
Comment (yurm04) [inline polaris.shopify.com/src/components/GlobalSearch/GlobalSearch.tsx:55]: matching the GA convention

File: polaris.shopify.com/src/components/GlobalSearch/GlobalSearch.tsx [hunk 4]
```diff
@@ -52,23 +52,14 @@ function captureSearchEvent(
...
-  const googleParams = {
-    event_category: resultRank > 0 ? 'engagement' : 'exit',
-    event_label: selectedResult,
-    value: resultRank,
-  };
-
   if (process.env.NODE_ENV === 'production') {
-    // i honestly have no idea which one of thes is the right set up for the google analytics version we have so let's try both
-    // and keep the one that works
     window.gtag('event', 'customSearch', customParams);
```
Comment (yurm04) [inline polaris.shopify.com/src/components/GlobalSearch/GlobalSearch.tsx:62]: this is the only event we need for now in search

**Review by laurkim — APPROVED** [review]

**Pull request merged**